### PR TITLE
Use UNIXSocket to pass file descriptor

### PR DESCRIPTION
**Pull request by xu-cheng**
@jacknagel It works. Thanks for the tip.

Github doesn't allow me reopen the original PR after I run `push --force`. So open a new one.

### CODE REVIEW

File: Library/Homebrew/formula_installer.rb [hunk 1]
```diff
@@ -492,10 +493,16 @@ def build
     end
 
     ignore_interrupts(:quietly) do # the child will receive the interrupt and marshal it back
+      begin
+        socket = server.accept_nonblock
+      rescue Errno::EAGAIN, Errno::EWOULDBLOCK, Errno::ECONNABORTED, Errno::EPROTO, Errno::EINTR
+        retry unless Process.waitpid(pid, Process::WNOHANG)
+      end
+      socket.send_io(write) unless socket.nil?
```
Comment (jacknagel) [inline Library/Homebrew/formula_installer.rb]: Should be able to do

```rb
begin
  # ...
rescue # ...
  # ...
else
  socket.send_io(write)
end
```

and skip the nil check here.

Comment (xu-cheng) [inline Library/Homebrew/formula_installer.rb]: Updated. Though `socket.nil?` is still required in `Process.wait(pid)`

**Comment by xu-cheng** [comment]
Any more suggestion before I merge this?

**Comment by mikemcquaid** [comment]
LGTM :+1:

**Pull request closed without merging**